File: Python/30-连续数组的最大和.py
# -*- coding:utf-8 -*-
class Solution:
    def FindGreatestSumOfSubArray(self, array):
        # write code here
        if len(array) == 1:
            return array[0]
        res = array[0]
        cur = res if res > 0 else 0
        for i in range(1, len(array)):
            cur += array[i]
            if cur > res:
                res = cur
            if cur < 0:
                cur = 0
        return res

File: C++/30-连续数组的最大和.cpp
class Solution
{
public:
    int FindGreatestSumOfSubArray(vector<int> array)
    {
        int res = 0x80000000;
        int cur = 0;
        for (int i = 0; i < array.size(); ++i)
        {
            cur += array[i];
            if (cur > res)
                res = cur;
            if (cur <= 0)
                cur = 0;
        }
        return res;
    }
};